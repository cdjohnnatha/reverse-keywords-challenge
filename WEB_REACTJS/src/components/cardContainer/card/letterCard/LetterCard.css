.card-class {
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
  transition: 0.3s;
  border-radius: 5px;
  padding: 1em;
  border: 1px solid lightblue;
  cursor: pointer;
}

.card-class:hover {
  background-color: grey
}

.flex-container {
  display: flex;
  justify-content: center;
}

.box-margin {
  margin: 1em 0.2em 1em 0.2em;
}

.green-bg {
  background-color: lightgreen;
}

.red-bg {
  background-color: lightcoral;
}