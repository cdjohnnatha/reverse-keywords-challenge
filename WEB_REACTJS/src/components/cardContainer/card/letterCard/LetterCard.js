import React from 'react';
import { string, func, number, bool } from 'prop-types';
import './LetterCard.css';

const LetterCard = ({
  letter,
  letterIndexPosition,
  onSingleLetterClickHandler,
  isFromWordHistory,
  cardHistoryIndex,
}) => {
  let classList = 'card-class flex-container box-margin';
  if (isFromWordHistory) classList += ' red-bg';
  else classList += ' green-bg';

  return (
    <div
      className={classList}
      onClick={() =>
        onSingleLetterClickHandler(letterIndexPosition, cardHistoryIndex)
      }
    >
      {letter}
    </div>
  );
};

LetterCard.propTypes = {
  letter: string.isRequired,
  onSingleLetterClickHandler: func.isRequired,
  letterIndexPosition: number.isRequired,
  cardHistoryIndex: number,
  isFromWordHistory: bool.isRequired,
};

LetterCard.defaultProps = {
  cardHistoryIndex: null,
};

export default LetterCard;
