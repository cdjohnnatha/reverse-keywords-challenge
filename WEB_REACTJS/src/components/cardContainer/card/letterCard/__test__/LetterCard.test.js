import React from 'react';
import { create, act } from 'react-test-renderer';
import { mount } from 'enzyme';
import LetterCard from '../LetterCard';
import '../../../../../../jest.setup';

const onSingleLetterClickHandler = () => {};

describe('Test LetterCard component', () => {
  test('snapshot renders', () => {
    const component = create(
      <LetterCard
        letter="K"
        letterIndexPosition={0}
        onSingleLetterClickHandler={onSingleLetterClickHandler}
        isFromWordHistory={false}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('LetterCard component is a div with one letter', () => {
    const component = create(
      <LetterCard
        letter="K"
        letterIndexPosition={0}
        onSingleLetterClickHandler={onSingleLetterClickHandler}
        isFromWordHistory={false}
      />
    ).root;
    const inputField = component.findByType('div');

    expect(inputField.type).toBe('div');
    expect(inputField.props.children).toBe('K');
  });
  test('LetterCard component should be green with considered as recentInput', () => {
    const component = mount(
      <LetterCard
        letter="K"
        letterIndexPosition={0}
        onSingleLetterClickHandler={onSingleLetterClickHandler}
        isFromWordHistory={false}
      />
    );

    expect(component.find('.green-bg').length).toBe(1);
    expect(component.find('.red-bg').length).toBe(0);
  });
  test('LetterCard component should be red with considered as history', () => {
    const component = mount(
      <LetterCard
        letter="K"
        letterIndexPosition={0}
        onSingleLetterClickHandler={onSingleLetterClickHandler}
        isFromWordHistory
      />
    );

    expect(component.find('.green-bg').length).toBe(0);
    expect(component.find('.red-bg').length).toBe(1);
  });
});
