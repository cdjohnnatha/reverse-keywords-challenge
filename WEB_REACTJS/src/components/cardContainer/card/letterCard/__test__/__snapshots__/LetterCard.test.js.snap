// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LetterCard component snapshot renders 1`] = `
<div
  className="card-class flex-container box-margin green-bg"
  onClick={[Function]}
>
  K
</div>
`;
